### PR TITLE
refactor(website): use starlight

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

This PR changes the website to use starlight!

## Test Plan

Check the rendered website

https://biome-website-2b5ui2x80-ematipico.vercel.app/

### CODE REVIEW

**Comment by Conaclos** [comment]
Which are the advantages of using Starlight instead of the current solution?

File: Cargo.lock [hunk 1]
```diff
@@ -3638,6 +3647,7 @@ dependencies = [
 name = "xtask_lintdoc"
 version = "0.0.0"
 dependencies = [
+ "convert_case",
```
Comment (Conaclos) [inline Cargo.lock:3650]: Do you think we could use our [in-house implementation](https://github.com/biomejs/biome/blob/main/crates/rome_js_analyze/src/utils/case.rs)?

**Comment by ematipico** [comment]
> Which are the advantages of using Starlight instead of the current solution?

Sorry, I meant to write a better description later.

There are many reasons:
- it has support for various features we need, out of the box: search (we are still using Rome's Algolia), translations, banners for announcements;
- they plan to develop versioned docs, which will fix your documentation issues;
- it's actively maintained;
- it removes a lot of **maintenance** **burden** from us;
- it provides better navigation, the one on the right: on mobile it's impossible to navigate currently;
- we can still use our own styles, in fact the playground is still there;

File: Cargo.lock (hunk 1 above)
Comment (ematipico) [reply Cargo.lock:3650]: I didn't want to bother, the crate is used for a tooling internal script

**Comment by ematipico** [comment]
What do you think @Conaclos? Do you like the website? Should we add something else? Is there anything missing?

**Comment by Conaclos** [comment]
Looks nice!

Some questions/suggestions:

- Is there a way to better integrate the playground? I mean re-using the header of the website?
- On the homepage, you use a webp format for the logo. Why not use svg directly?
- Could we add a way to easily go to the docs / blog and playground without first clicking on "Get Started"? I was a bit lost the first time...
- On the homepage, a thunderbolt emoji ⚡ for "Fast"  could be better.
- What about using "Formats, lints, and more in a fraction of a second" as slogan instead of "Format, lint, and more with a fast and reliable toolchain."?

**Comment by ematipico** [comment]
Updated the description with the latest deploy

**Review by Conaclos — APPROVED** [review]

**Pull request merged**